ddtrace/tracer/span.go

```diff
@@ -363,8 +363,7 @@ func (s *span) finish(finishTime int64) {
 	keep := true
 	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
```

Comment:
Couldn't you just have done this
```suggestion
	if t, ok := internal.GetGlobalTracer().(*tracer); ok && t.config.HasFeature("discovery") {
```
And avoid creating the `canDropP0s` and `canComputeStats` functions?